### PR TITLE
 fix compatibility for newer versions  & add servo support

### PR DESCRIPTION
**Pull request by vaishnavachath**
replaced some deprecated functions
modified binding.gyp to choose sources based on rc_version
added servo support , used @chnewso 's fork for reference
tested on:
 BeagleBone Blue BeagleBoard.org Debian Image 2018-03-05
Linux beaglebone 4.9.105-ti-r113
rc_version : 0.4.4

### CODE REVIEW

**Comment by vaishnavachath** [comment]
https://github.com/jadonk/node-roboticscape/issues/4
https://github.com/jadonk/node-roboticscape/issues/3

File: binding.gyp [hunk 1]
```diff
@@ -2,7 +2,21 @@
     "targets": [
         {
             "target_name": "roboticscape",
-            "sources": [ "rc-node-bindings.cc" ],
+	    "variables": {
```
Comment (jadonk) [inline binding.gyp]: Indentation looks different.

File: binding.gyp [hunk 2]
```diff
@@ -2,7 +2,21 @@
...
+		   'RC_version': '<!(rc_version)'
+            },
+            "conditions": [
```
Comment (jadonk) [inline binding.gyp]: Please use 0.4.4 and obsolete use for the older library. In fact, James has recently made a 1.0.0 release: https://github.com/StrawsonDesign/librobotcontrol/releases/tag/V1.0.0

It should be 0.4.4 compatible, I hope.

File: rc-node-bindings041.cc [hunk 3]
```diff
@@ -0,0 +1,504 @@
+#include <nan.h>
```
Comment (jadonk) [inline rc-node-bindings041.cc]: Please don't create a new file. We can obsolete support for the old API. I'd like to see what changed in this file.

**Review by jadonk — CHANGES_REQUESTED** [review]
Until I can see what changed in the source, I don't want to approve the merge yet. Go ahead and remove support for the old API.

**Comment by vaishnavachath** [comment]
submiited https://github.com/jadonk/node-roboticscape/pull/6 after making the recommended changes

**Pull request closed without merging**